### PR TITLE
chore: update C, improve preproc_arg highlights

### PR DESCRIPTION
**Pull request by amaanq**

### CODE REVIEW

**Comment by clason** [comment]
Could you try to avoid unrelated reformatting? Makes it harder to evaluate the diff.

**Comment by amaanq** [comment]
for sure, updated

File: queries/c/highlights.scm [hunk 1]
```diff
@@ -157,13 +158,17 @@
...
+  (#lua-match? @constant "^[A-Z][A-Z0-9_]+$"))
 (enumerator
   name: (identifier) @constant)
 (case_statement
   value: (identifier) @constant)
 
 ((identifier) @constant.builtin
     (#any-of? @constant.builtin "stderr" "stdin" "stdout"))
+(preproc_def (preproc_arg) @constant.builtin
+  (#any-of? @constant.builtin "stderr" "stdin" "stdout"))
```
Comment (clason) [inline queries/c/highlights.scm:171]: Would it make sense to keep these together?

(I guess this is a general question about structuring queries -- where not constrained by precedence -- by node or by capture.)

**Review by clason — APPROVED** [review]

File: queries/c/highlights.scm (hunk 1 above)
Comment (amaanq) [reply queries/c/highlights.scm:171]: not sure I follow - are you asking does it make sense to also check for these in a preproc_arg? I'd like to say yes if so

Comment (clason) [reply queries/c/highlights.scm:171]: No, I just mean there's three different stanzas for the same node with different captures, in three different places (where the same capture is applied to different nodes).

Comment (amaanq) [reply queries/c/highlights.scm:171]: ah, yeah we could definitely keep these together for this one

Comment (clason) [reply queries/c/highlights.scm:171]: Just a thought; not a deal-breaker either way.

**Pull request merged**